bitcoin/src/blockdata/weight.rs

```diff
@@ -54,6 +54,21 @@ impl Weight {
         vb.checked_mul(Self::WITNESS_SCALE_FACTOR).map(Weight::from_wu)
     }
 
+    /// Constructs `Weight` from virtual bytes in const context.
+    pub const fn from_vb_const(vb: u64) -> Weight {
```

Comment:
Sorry for late review, I was very sick. NACK this name. Should be something like `panicking_from_vb` or `asserted_from_vb` (do we use this in other parts of the crate?). Also the documentation is missing panic section.

Reply:
Sure, we can change the name, I'm not particular about that.  I left panic out actually since it doesn't panic during run-time like our other documented panic functions, but instead panics during compile time since it's const.  I think it confusing to document them both the same..

Reply:
That's not right. There's no guarantee someone won't use it non-const context and in that case it's a runtime panic.

Reply:
Hmm good point.  Could we just remove the non const `from_vb` and rename this const version `from_vb` then?

Reply:
No, `from_vb` returns an `Option` for a reason. Ideally `unwrap` would be available in const context but unfortunately it's not yet.

And I just came up with the name `from_vb_unwrap` which I really like.

Reply:
cool, here is the PR https://github.com/rust-bitcoin/rust-bitcoin/pull/2228